src/codegen/llvm.zig

```diff
@@ -145,6 +148,7 @@ pub fn targetTriple(allocator: *Allocator, target: std.Target) ![:0]u8 {
 
 pub const LLVMIRModule = struct {
     module: *Module,
+    decl: ?*Module.Decl = null,
```

Comment:
This field should be below the `// TODO: The fields below should ...` comment
(or better yet, let's take the opportunity here to do the suggested reworking and introduce a per-function struct for the state that mutates, instead of mutating `LLVMIRModule`.

Reply:
I have a WIP branch where I did this, still have to upstream that

Reply:
Ah, Ill move it into todo for now, and we can do the refactor when you do it.